### PR TITLE
BTFS-1968 BTFS-1969 BTFS-1971

### PR DESCRIPTION
**Pull request by laipogo**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior?** (You can also refer to a JIRA ticket here)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **What dependencies / modules need to be updated as pre-requisites?** (Include relevant PR links here)

* **Description of changes**

---

* **Please check if the PR fulfills these requirements**
- [ ] The subject of this PR contains a JIRA ticket BTFS-xxxx (N/A for community PRs)
- [ ] PR description and commit messages are [good and meaningful](https://chris.beams.io/posts/git-commit/)
- [ ] Unit tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **Please make sure the following procedures have been applied before requesting reviewers**
- [ ] Code changes closely follow [Go Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments)
- [ ] Code has been *rebased* against latest master (`git merge` not recommended, unless you know what you are doing)
- [ ] Code changes have run through `go fmt`
- [ ] Code changes have run through `go mod tidy`
- [ ] All unit tests passed locally (`make test_go_test`)

### CODE REVIEW

File: core/commands/wallet.go [hunk 1]
```diff
@@ -173,3 +180,136 @@ var walletBalanceCmd = &cmds.Command{
...
+	Helptext: cmds.HelpText{
+		Tagline:          "BTFS wallet password",
+		ShortDescription: "set password for BTFS wallet",
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("password", true, false, "password of BTFS wallet."),
+	},
+	Options: []cmds.Option{},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		cfg, err := cmdenv.GetConfig(env)
```
Comment (taiyangc) [inline core/commands/wallet.go:194]: This needs to be protected by a new wallet init flag (under config.UI section), if already init, cannot set password again.

File: core/commands/wallet.go [hunk 2]
```diff
@@ -173,3 +180,136 @@ var walletBalanceCmd = &cmds.Command{
...
+		ShortDescription: "get keys of BTFS wallet",
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("password", false, false, "password of BTFS wallet."),
+	},
+	Options: []cmds.Option{
+		cmds.StringOption(passwordOptionName, "p", "password of BTFS wallet"),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		cfg, err := cmdenv.GetConfig(env)
```
Comment (taiyangc) [inline core/commands/wallet.go:230]: Same, protected by init flag.

File: core/commands/wallet.go [hunk 3]
```diff
@@ -173,3 +180,136 @@ var walletBalanceCmd = &cmds.Command{
...
+
+const passwordOptionName = "password"
+
+var walletKeysCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline:          "BTFS wallet keys",
+		ShortDescription: "get keys of BTFS wallet",
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("password", false, false, "password of BTFS wallet."),
```
Comment (taiyangc) [inline core/commands/wallet.go:224]: should be required.

File: core/commands/wallet.go [hunk 4]
```diff
@@ -173,3 +180,136 @@ var walletBalanceCmd = &cmds.Command{
...
+
+type Transaction struct {
+	Datetime time.Time
+	Amount   int64
+	From     string
+	To       string
+	Status   string
+}
+
+var walletStatusCmd = &cmds.Command{
```
Comment (taiyangc) [inline core/commands/wallet.go:290]: No need for this command. Use an actual flag under `config.UI.Wallet.Initialized`

File: core/commands/wallet.go (hunk 3 above)
Comment (laipogo) [reply core/commands/wallet.go:224]: delete arguments, use Options instead.

File: core/commands/config.go [hunk 5]
```diff
@@ -172,9 +172,11 @@ NOTE: For security reasons, this command will omit your private key. If you woul
 			return err
 		}
 
-		err = scrubValue(cfg, []string{config.IdentityTag, config.PrivKeyTag})
-		if err != nil {
-			return err
+		for _, k := range []string{config.PrivKeyTag, config.MnemonicTag, config.Password} {
```
Comment (taiyangc) [inline core/commands/config.go:175]: It should be called PasswordTag to avoid confusion....

**Pull request merged**